returnn/torch/engine.py

```diff
@@ -380,6 +384,13 @@ def train_epoch(self):
                     torch.distributed.all_reduce(_has_data, op=torch.distributed.ReduceOp.MIN)
                 if not _has_data[0]:
                     break
+
+                keys_w_seq_len = [k for k in extern_data_raw if f"{k}:seq_len" in extern_data_raw]
+                data_len = NumbersDict({k: torch.sum(extern_data_raw[f"{k}:seq_len"]).item() for k in keys_w_seq_len})
+                data_space = NumbersDict({k: torch.prod(extern_data_raw[k].shape[:2]).item() for k in keys_w_seq_len})
```

Comment:
```suggestion
                data_len = NumbersDict({k: sum(extern_data_raw[f"{k}:seq_len"]) for k in keys_w_seq_len})
                data_space = NumbersDict({k: prod(extern_data_raw[k].shape[:2]) for k in keys_w_seq_len})
```
For `prod`, do `from returnn.util.basic import prod`.